### PR TITLE
Feat/text field tweaks

### PR DESCRIPTION
**Pull request by stefanmarbun12**
- Prop to show a copy to clipboard button for the text field
- Prop to validate the value of the text input and show the corresponding icons

### CODE REVIEW

File: src/components/TextField.tsx [hunk 1]
```diff
@@ -9,11 +9,17 @@ import {
   FormHelperText,
   FormErrorMessage,
   Divider,
+  IconButton,
+  InputLeftElement,
+  Spinner,
 } from '@chakra-ui/react';
+import { CopyIcon, CheckIcon, WarningIcon } from '@chakra-ui/icons';
+import CopyToClipboard from 'react-copy-to-clipboard';
```
Comment (tomanagle) [inline src/components/TextField.tsx:17]: Why do we need a library for this? Can use `navigator.clipboard.writeText("boo")`

File: src/components/TextField.tsx [hunk 2]
```diff
@@ -39,19 +45,38 @@ export const TextField: FC<FieldProps<TextFieldSchema>> = ({
...
 
-  const { register, control } = useFormContext();
+  const { register, control, watch } = useFormContext();
 
   const errorMessage = useErrorMessage(name, label);
 
   const values = useWatch({
     control,
   });
+  const value = watch(name);
```
Comment (tomanagle) [inline src/components/TextField.tsx:64]: If we have values above, from useWatch, we should be able to derive the value given the name from that

File: src/components/TextField.tsx [hunk 3]
```diff
@@ -39,19 +45,38 @@ export const TextField: FC<FieldProps<TextFieldSchema>> = ({
...
+  const debouncedValue = useDebounce(value, 500);
+
+  const [valid, setValid] = useState(true);
+
+  useEffect(() => {
+    const validate = async () => {
```
Comment (tomanagle) [inline src/components/TextField.tsx:70]: What happens if `validateOnChange` is undefined?

File: src/components/TextField.tsx (hunk 1 above)
Comment (stefanmarbun12) [reply src/components/TextField.tsx:17]: Whoops it's the same one used in platform portal, and now I'm wondering why that library needed to exist in the first place 🤔

File: src/stories/Sandbox.stories.tsx [hunk 4]
```diff
@@ -154,6 +154,10 @@ Sandbox.args = {
       isRequired: true,
       defaultValue: 'compare',
       divideAfter: true,
+      copyToClipboard: true,
+      validateOnChange: async (value) => {
```
Comment (tomanagle) [inline src/stories/Sandbox.stories.tsx:158]: Do we want to be able to define the debounce time here? If so, could do something like

`
asyncValidator:{
timeout: number,
handler: async () => {}
} 
`

File: src/components/TextField.tsx (hunk 3 above)
Comment (stefanmarbun12) [reply src/components/TextField.tsx:70]: Absolutely nothing haha, `valid` will still be true and the icons are shown if `validateOnChange` is defined anyway

File: src/components/TextField.tsx (hunk 1 above)
Comment (tomanagle) [reply src/components/TextField.tsx:17]: I guess there probably wasn't a good web api for it at some point, but I think it's redundant now

**Review by tomanagle — APPROVED** [review]

File: src/components/TextField.tsx (hunk 3 above)
Comment (tomanagle) [reply src/components/TextField.tsx:70]: Should we wrap it all if an `if validateOnChange`?

File: src/stories/Sandbox.stories.tsx (hunk 4 above)
Comment (stefanmarbun12) [reply src/stories/Sandbox.stories.tsx:158]: Sounds good! I'll just move anything to do with the validation into an `inputValidation` prop

**Pull request merged**